### PR TITLE
[EOS-18405] Integrate refresh machine_id states with setup provisioner API

### PR DESCRIPTION
**Pull request by mazinamdar**
Signed-off-by: mazinamdar mazhar.inamdar@seagate.com

Grains data is not getting refreshed within sls files if we call init.sls for machine_id states.
so applied states separately.

### CODE REVIEW

File: api/python/provisioner/commands/setup_provisioner.py [hunk 1]
```diff
@@ -1803,6 +1803,18 @@ def _run(self, nodes, **kwargs):  # noqa: C901 FIXME
             ), targets=run_args.primary.minion_id
         )
 
+        # Grains data is not getting refreshed within sls files
+        # if we call init.sls for machine_id states.
+        logger.info("Refresh machine id on the system")
+        for state in [
+            'components.provisioner.config.machine_id.refresh_machine_id',
```
Comment (andkononykhin2) [inline api/python/provisioner/commands/setup_provisioner.py:1810]: @mazinamdar can't we just apply only `components.provisioner.config.machine_id`?

**Review by andkononykhin2 — COMMENTED** [review]
@mazinamdar Looks good. Please check one question. Thanks

File: api/python/provisioner/commands/setup_provisioner.py (hunk 1 above)
Comment (mazinamdar) [reply api/python/provisioner/commands/setup_provisioner.py:1810]: I am facing same issue of grains refresh if I run `components.provisioner.config.machine_id`  so we need to run both states separately.

Comment (andkononykhin2) [reply api/python/provisioner/commands/setup_provisioner.py:1810]: Got it. Thanks

**Review by andkononykhin2 — APPROVED** [review]

**Review by ypise — APPROVED** [review]

**Pull request merged**